h2o-algos/src/main/java/hex/mli/loco/LeaveOneCovarOut.java

```diff
@@ -0,0 +1,245 @@
+package hex.mli.loco;
+
+import hex.Model;
+import hex.ModelCategory;
+import water.MRTask;
+import water.ParallelizationTask;
+import water.exceptions.H2OIllegalArgumentException;
+import water.fvec.Chunk;
+import water.fvec.Frame;
+import water.fvec.NewChunk;
+import water.fvec.Vec;
+import water.DKV;
+import water.H2O;
+import water.util.Log;
+import water.Iced;
+import water.Job;
+import water.rapids.ast.prims.reducers.AstMedian;
+import hex.quantile.QuantileModel.CombineMethod;
+import water.Key;
+
+/**
+ * Leave One Covariate Out (LOCO)
+ *
+ * Calculates row-wise variable importance's by re-scoring a trained supervised model and measuring the impact of setting
+ * each variable to missing or it’s most central value(mean or median & mode for categorical's)
+ *
+ */
+public class LeaveOneCovarOut extends Iced {
+
+    /**
+     * Conduct Leave One Covariate Out (LOCO) given a model, frame, job, and replacement value
+     * @param m Supervised H2O model
+     * @param fr H2O Frame to score
+     * @param job Job to keep track of in terms of progress
+     * @param replaceVal Value to replace column by when conducting LOCO ("mean" or "median"). Default behavior is to setting to NA
+     * @param frameKey Key of final Leave One Covariate Out(LOCO) Frame.
+     * @return  An H2OFrame displaying the base prediction (model scored with all predictors) and the difference in predictions
+     *          when variables are dropped/replaced. The difference displayed is the base prediction substracted from
+     *          the new prediction (when a variable is dropped/replaced with mean/median/mode) for binomial classification
+     *          and regression problems. For multinomial problems, the sum of the absolute value of differences across classes
+     *          is calculated per column dropped/replaced.
+     */
+    public static Frame leaveOneCovarOut(Model m, Frame fr, Job job, String replaceVal, Key frameKey){
+
+        Frame locoAnalysisFrame = new Frame();
+        if(m._output.getModelCategory() != ModelCategory.Multinomial) {
```

Comment:
Minor/Doc: i would comment shortly each block - it will become more readable ...